units/host_monitoring_idle_termination_test.go

```diff
@@ -175,3 +192,194 @@ func TestFlaggingIdleHosts(t *testing.T) {
 		})
 	})
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// Testing with reference distro ids that are not present in the 'distro' collection in the database
+//
+
+func TestFlaggingIdleHostsWithMissingDistroIDs(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	testConfig := testutil.TestConfig()
+	db.SetGlobalSessionProvider(testConfig.SessionFactory())
+
+	env := evergreen.GetEnvironment()
+
+	Convey("When flagging idle hosts to be terminated", t, func() {
+
+		// reset the db
+		testutil.HandleTestingErr(db.ClearCollections(distro.Collection),
+			t, "error clearing distros collection")
+		testutil.HandleTestingErr(db.ClearCollections(host.Collection),
+			t, "error clearing hosts collection")
+
+		// insert two reference distro.Distro
+		distro1 := distro.Distro{
+			Id: "distro1",
+			PlannerSettings: distro.PlannerSettings{
+				MinimumHosts: 2,
+			},
+		}
+		distro2 := distro.Distro{
+			Id: "distro2",
+			PlannerSettings: distro.PlannerSettings{
+				MinimumHosts: 1,
+			},
+		}
+
+		testutil.HandleTestingErr(distro1.Insert(), t, "error inserting distro")
+		testutil.HandleTestingErr(distro2.Insert(), t, "error inserting distro")
+
+		Convey("Add some hosts with referenced host.Distro that do not exist in the distro collection ", func() {
```

Comment:
would it be possible to take this out of goconvey?

Reply:
Yes